### PR TITLE
Make InputDeviceId's constructor constexpr.

### PR DESCRIPTION
**Pull request by bosnichd**
See also https://github.com/o3de/o3de/pull/4220

Signed-off-by: bosnichd <bosnichd@amazon.com>

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/Input/Devices/InputDeviceId.cpp [hunk 1]
```diff
@@ -30,37 +30,10 @@ namespace AzFramework
...
-    InputDeviceId& InputDeviceId::operator=(const InputDeviceId& other)
-    {
-        azstrcpy(m_name, NAME_BUFFER_SIZE, other.m_name);
-        m_crc32 = other.m_crc32;
-        m_index = other.m_index;
-        return *this;
-    }
-
     ////////////////////////////////////////////////////////////////////////////////////////////////
     const char* InputDeviceId::GetName() const
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzFramework/AzFramework/Input/Devices/InputDeviceId.cpp:34]: You can make this `constexpr` as well.

File: Code/Framework/AzFramework/Tests/InputTests.cpp [hunk 2]
```diff
@@ -48,6 +48,22 @@ namespace InputUnitTests
         AZStd::unique_ptr<InputSystemComponent> m_inputSystemComponent;
     };
 
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    TEST_F(InputTest, InputChannelId_ConstExpression_CopyConstructorSuccessfull)
+    {
+        constexpr InputChannelId testInputChannelId1("TestInputChannelId");
+        constexpr InputChannelId testInputChannelId2(testInputChannelId1);
+        EXPECT_EQ(testInputChannelId1, testInputChannelId2);
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzFramework/Tests/InputTests.cpp:56]: To truly test for `constexpr-ness`
Add a `static_assert` here.
i.e
`static_assert(testInputChannelId1 == testInputChannelId2);`

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Framework/AzFramework/AzFramework/Input/Devices/InputDeviceId.cpp (hunk 1 above)
Comment (bosnichd) [reply Code/Framework/AzFramework/AzFramework/Input/Devices/InputDeviceId.cpp]: Good suggestion, done (along with the other 'Get' and comparison member functions).

File: Code/Framework/AzFramework/Tests/InputTests.cpp (hunk 2 above)
Comment (bosnichd) [reply Code/Framework/AzFramework/Tests/InputTests.cpp]: Done (had to make the equality comparison function constexpr also).

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by AMZN-ScottR — APPROVED** [review]

**Pull request merged**